python/semantic_kernel/connectors/memory/azure_ai_search/utils.py

```diff
@@ -0,0 +1,46 @@
+# Copyright (c) Microsoft. All rights reserved.
+
+from typing import TYPE_CHECKING
+
+from azure.core.credentials import AzureKeyCredential, TokenCredential
+from azure.search.documents.indexes.aio import SearchIndexClient
+
+from semantic_kernel.connectors.memory.azure_ai_search.azure_ai_search_settings import AzureAISearchSettings
+from semantic_kernel.connectors.telemetry import APP_INFO, prepend_semantic_kernel_to_user_agent
+from semantic_kernel.exceptions import ServiceInitializationError
+
+if TYPE_CHECKING:
+    from azure.core.credentials_async import AsyncTokenCredential
+
+
+def get_search_index_async_client(
+    azure_ai_search_settings: AzureAISearchSettings,
+    azure_credential: AzureKeyCredential | None = None,
+    token_credential: "AsyncTokenCredential | TokenCredential | None" = None,
```

Comment:
Should `token_credential` also support `AsyncTokenCredential` at runtime here (and for related methods such as `azure_ai_search_vector_record_store`)? It looks like the methods that call `get_search_index_async_client` only support `TokenCredential`, but `get_search_index_async_client` technically does support `AsyncTokenCredential` unless I'm misunderstanding this.